lib/puppetserver/ca/x509_loader.rb

```diff
@@ -5,31 +5,61 @@ module Ca
     # Load, validate, and store x509 objects needed by the Puppet Server CA.
     class X509Loader
 
-      attr_reader :errors, :certs, :key, :crls
+      attr_reader :errors, :certs, :cert, :key, :crls, :crl
 
       def initialize(bundle_path, key_path, chain_path)
         @errors = []
 
         @certs = load_certs(bundle_path)
         @key = load_key(key_path)
         @crls = load_crls(chain_path)
+        @cert = find_signing_cert
+        @crl = find_leaf_crl
```

Comment:
Would it make sense to pass the needed objects into these methods?

Reply:
i've seen competing styles here including intentional refactors to move to using instance variables... i'm fine either way and probably prefer passing them in

Reply:
I do think it's okay to use instance variables in an object like a loader if you have clear lifecycles

Reply:
maybe not worth it?